core/src/toga/colors.py

```diff
@@ -1,2 +1,52 @@
-# Use the Travertino color definitions as-is
-from travertino.colors import *  # noqa: F401, F403
+from typing import TYPE_CHECKING
+
+# Color, rgb, and hsl need to be explicitly imported in order for mkdocstrings to see
+# them. However, we also want to import all 148 named colors, and it seems silly to
+# list them here individually.
+from travertino.colors import *  # noqa: F401, F403, I001
+from travertino.colors import Color, hsl, rgb  # noqa: F401, I001
+
+if TYPE_CHECKING:
+    from typing import TypeAlias
+
+    ColorT: TypeAlias = Color | str
+    """
+    Toga's color APIs accept:
+
+    - An instance of Toga's [`rgb`][toga.colors.rgb] or [`hsl`][toga.colors.hsl] class.
+    - The name of a [CSS named color](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/named-color).
+      Strings of all available color names are available as constants. These constants
+      are named the same as their value, except in all caps, as is the convention for
+      Python constants. In other words, `toga.colors.ORANGE == "orange"`.
+    - A string representing the color as [hexadecimal RGB or RGBA](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Values/hex-color).
+
+    The following color definitions would all be equivalent:
+
+    ```python
+    import toga
+    from toga.colors import hsl, rgb, YELLOW
+
+    toga.Box(color=rgb(255, 255, 0))
+    toga.Box(color=hsl(60, 1, .5))
+    toga.Box(color="yellow")
+    toga.Box(color=YELLOW)
+    toga.Box(color="#FF0")
+    toga.Box(color="#FFFF00")
```

Comment:
It occurred to me that using a color that doesn't reproduce the same values in the RGB channel might be helpful. And it turns out REBECCAPURPLE does that.
```suggestion
    from toga.colors import hsl, rgb, REBECCAPURPLE

    toga.Box(color=rgb(102, 51, 153))
    toga.Box(color=hsl(270, 0.5, 0.4))
    toga.Box(color="rebeccapurple")
    toga.Box(color=REBECCAPURPLE)
    toga.Box(color="#639")
    toga.Box(color="#663399")
```